bundles/org.openhab.binding.nikobus/src/main/java/org/openhab/binding/nikobus/internal/handler/NikobusModuleHandler.java

```diff
@@ -0,0 +1,247 @@
+            return;
+        }
+
+        updateGroup(SwitchModuleGroup.mapFromChannel(channelUID));
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        synchronized (linkedChannels) {
+            linkedChannels.add(channelUID);
+        }
+        super.channelLinked(channelUID);
+    }
+
+    @Override
+    public void channelUnlinked(ChannelUID channelUID) {
+        synchronized (linkedChannels) {
+            linkedChannels.remove(channelUID);
+        }
+        super.channelUnlinked(channelUID);
+    }
+
+    public void refreshModule() {
+        Set<SwitchModuleGroup> groups = new HashSet<>();
+        synchronized (linkedChannels) {
+            for (ChannelUID channelUID : linkedChannels) {
+                groups.add(SwitchModuleGroup.mapFromChannel(channelUID));
+            }
+        }
+
+        if (groups.isEmpty()) {
+            logger.debug("Nothing to refresh for '{}'", thing.getUID());
+            return;
+        }
+
+        logger.debug("Refreshing {} - {}", thing.getUID(), groups);
+
+        for (SwitchModuleGroup group : groups) {
+            updateGroup(group);
+        }
+    }
+
+    public void requestStatus(SwitchModuleGroup group) {
+        updateGroup(group);
+    }
+
+    private void updateGroup(SwitchModuleGroup group) {
+        synchronized (pendingRefresh) {
+            if (pendingRefresh.contains(group)) {
+                logger.debug("Refresh already scheduled for group {} of module '{}'", group, getAddress());
+                return;
+            }
+
+            pendingRefresh.add(group);
+        }
+
+        logger.debug("Refreshing group {} of switch module '{}'", group, getAddress());
+
+        NikobusPcLinkHandler pcLink = getPcLink();
+        if (pcLink != null) {
+            NikobusCommand command = SwitchModuleCommandFactory.createReadCommand(getAddress(), group,
+                    result -> processStatusUpdate(result, group));
+            pcLink.sendCommand(command);
+        }
+    }
+
+    private void processStatusUpdate(NikobusCommand.Result result, SwitchModuleGroup group) {
+        try {
+            String responsePayload = result.get();
+
+            logger.debug("processStatusUpdate '{}' for group {} in module '{}'", responsePayload, group, getAddress());
+
+            if (thing.getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+            // Update channel's statuses based on response.
+            for (int i = 0; i < group.getCount(); i++) {
+                String channelId = CHANNEL_OUTPUT_PREFIX + (i + group.getOffset());
+                String responseDigits = responsePayload.substring(9 + (i * 2), 11 + (i * 2));
+
+                int value = Integer.parseInt(responseDigits, 16);
+
+                updateStateAndCacheValue(channelId, value);
+            }
+        } catch (Exception e) {
```

Comment:
Can this be replaced by catching the specific checked exception and/or `RuntimException`?

Reply:
This is actually a consequence of using `Callable<V>` where `V call() throws Exception` ... not sure if there a different approach suggested i.e. not using `Callable` but rather custom or some other impl?